### PR TITLE
EDUCATOR-4979 update models to add team workflows

### PR DESCRIPTION
**Pull request by Andytr1**
Jira: [Educator-4979](https://openedx.atlassian.net/browse/EDUCATOR-4979)
Supporting teams project. Added `TeamStaffWorkflow` and `TeamAssessmentWorkflow` models.

- [ ] Make sure your PR updates the version number in ``setup.py`` and ``package.json``
- [ ] Run ``make javascript sass`` if your changes include JS/CSS changes.
- [ ] Get a green Travis build for this PR
- [ ] Merge to master
- [ ] Create a release tag on GitHub https://github.com/edx/edx-ora2/releases
- [ ] Manually inspect diff at https://github.com/edx/edx-ora2/compare/0.x.n-1...0.x.n, email contributors
- [ ] Create PR to update `requirements/edx/{github.in,base.txt,development.txt,testing.txt}` in `edx-platform`.
- [ ] If manual testing of the changes against edx-platform is desired, create a sandbox (or use devstack).
- [ ] Get green build on the edx-platform PR, merge.
- [ ] Consider any feature flags that must be changed.
- [ ] Once your code has been released to production, try to test it there, too.

### CODE REVIEW

File: openassessment/assessment/models/staff.py [hunk 1]
```diff
@@ -151,3 +151,10 @@ def close_active_assessment(self, assessment, scorer_id):
         self.scorer_id = scorer_id
         self.grading_completed_at = now()
         self.save()
+
+
+class TeamStaffWorkflow(StaffWorkflow):
+    """
+    Extends the StafWorkflow to be used for team based assessments.
+    """
+    team_submission_uuid = models.CharField(max_length=128, unique=True, null=False)
```
Comment (nsprenkle) [inline openassessment/assessment/models/staff.py]: Why this choice for max length?

File: settings/base.py [hunk 2]
```diff
@@ -78,6 +78,10 @@
         'BACKEND': 'django.template.backends.django.DjangoTemplates',
         'APP_DIRS': True,
         'OPTIONS': {
+            'context_processors': [
+                'django.contrib.auth.context_processors.auth',
```
Comment (nsprenkle) [inline settings/base.py]: Why did these get added?

**Comment by nsprenkle** [comment]
Are there any test updates that need to happen?

File: settings/base.py (hunk 2 above)
Comment (Andytr1) [reply settings/base.py]: This is needed to allow me to run migrations from within the venv of edx-ora2.

**Comment by Andytr1** [comment]
> Are there any test updates that need to happen?

Nothing at this time. Since the new models have no logic and we aren't "exercising" them.

**Comment by jansenk** [comment]
Was the package-json on purpose? otherwise i'd remove

**Comment by Andytr1** [comment]
> Was the package-json on purpose? otherwise i'd remove

Yes, by design. Will keep an eye, though to see that it doesn't break other things.

**Comment by jansenk** [comment]
> > Was the package-json on purpose? otherwise i'd remove
> 
> Yes, by design. Will keep an eye, though to see that it doesn't break other things.

Ah I should have just looked at the diff. Makes sense.

**Pull request merged**